packages/ember-testing/lib/ext/application.js

```diff
@@ -76,7 +76,7 @@ EmberApplication.reopen({
 
     this.testing = true;
 
-    this.Router.reopen({
+    this.resolveRegistration('router:main').reopen({
```

Comment:
FWIW, this concern is definitely non-blocking. Feel free to land this when you are ready.  I just think we _can_ avoid the reopen, but I may be missing something (or just be wrong 😝).